control/tests/optimal_test.py

```diff
@@ -381,17 +381,8 @@ def test_optimal_logging(capsys):
 @pytest.mark.parametrize("fun, args, exception, match", [
     [opt.quadratic_cost, (np.zeros((2, 3)), np.eye(2)), ValueError,
      "Q matrix is the wrong shape"],
-    [opt.quadratic_cost, (np.eye(2), 1), ValueError,
+    [opt.quadratic_cost, (np.eye(2), np.eye(2, 3)), ValueError,
      "R matrix is the wrong shape"],
-])
-def test_constraint_constructor_errors(fun, args, exception, match):
-    """Test various error conditions for constraint constructors"""
-    sys = ct.ss2io(ct.rss(2, 2, 2))
-    with pytest.raises(exception, match=match):
-        fun(sys, *args)
-
-
-@pytest.mark.parametrize("fun, args, exception, match", [
```

Comment:
I see the double `def` has to be fixed, but what is with the scalar shape? Is this not an error?

Reply:
1x1(x1x1...) is special cased:

https://github.com/python-control/python-control/blob/f591c080142177e07e3393b4d73356ef0d761d92/control/optimal.py#L1121-L1124

though this is not explicitly documented:

https://github.com/python-control/python-control/blob/f591c080142177e07e3393b4d73356ef0d761d92/control/optimal.py#L1097-L1098